src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveSummaryService.kt

```diff
@@ -159,6 +162,20 @@ data class IepResult(
   val daysOnLevel: Int
 )
 
+fun daysOnLevel(iepDetails: List<IepDetail>): Int {
+  val today = LocalDate.now().atStartOfDay()
+
+  val earliestMatchingIepDetail = iepDetails.reduce { earliestMatchingIepDetail, iepDetail ->
```

Comment:
`reduce` throws an `UnsupportedOperationException` if the iterable is empty. it should be clear enough without needing a bespoke exception class.

a bespoke class would require this block to be nested in a try-catch that re-throws a different exception… kinda unnecessary overkill?

Reply:
yeah, if `reduce {}` already throw an exception when list is empty I think that's good enough, especially for a programming error that should never happen.